base/array.jl

```diff
@@ -255,14 +255,14 @@ else
 end
 _default_eltype{I,T}(::Type{Generator{I,Type{T}}}) = T
 
-_array_for(T, itr, ::HasLength) = Array(T, Int(length(itr)::Integer))
-_array_for(T, itr, ::HasShape) = Array(T, convert(Dims,size(itr)))
+_array_for(T, itr, ::HasLength) = Array{T,ndims(itr)}(Int(length(itr)::Integer))
+_array_for(T, itr, ::HasShape) = Array{T,ndims(itr)}(convert(Dims,size(itr)))
 
```

Comment:
I'd leave the first one as `Array{T,1}` (it is the only `Array{T,N}` that can have a single `Int` as argument). The second one is probably better just as `Array{T}` since the `ndims(itr)` will make it non inferable.